src/script.ts

```diff
@@ -1,53 +1,85 @@
-  const gl = canvas.getContext('webgl2');
-
-  // Create Shader and Material
-  const material = new BaseColorMaterial(Color.fromHex('#302459'));
-
-  // Create Mesh and MeshNode
-  // ! Debug test for model loading data
-  const test = await Model.fromURL(
-    'https://static.observableusercontent.com/files/c1fc0d2fbf2bed5669afae79d4c0e896701b9e7257924c92a873b376bb2e65d7c217aeb899c11088d648cf89535a89089cdabff9da336ba7e6a739dd5e20a5cf',
-    material,
-  );
-  const meshNodes = test.MeshNodes;
-
-  // Create a Scene and insert the MeshNode as the root
-  const scene = new Scene();
-  meshNodes.forEach((node) => scene.addNode(node));
-
-  // Create a Camera to view the scene with
-  const camera = new PerspectiveCamera();
-
-  // Create the renderer and begin rendering
-  const preRender = () => {
-    camera.aspect = gl.canvas.width / gl.canvas.height;
-    meshNodes.forEach((meshNode) => {
-      meshNode.localTransform.rotation = twgl.v3.add(
-        meshNode.localTransform.rotation,
-        [1.0, 1.0, 1.0],
-      );
+    const canvas = document.getElementById("canvas") as HTMLCanvasElement;
+    const gl = canvas.getContext("webgl2");
+
+    // Generate Map
+    const map = PLACEHOLDER_MAP;
+
+    // Create Shader and Material
+    const material = new BaseColorMaterial(Color.fromHex("#302459"));
+
+    // Create Mesh and MeshNode
+    // ! Debug test for model loading data
+    const test = await Model.fromURL(
+        "https://static.observableusercontent.com/files/c1fc0d2fbf2bed5669afae79d4c0e896701b9e7257924c92a873b376bb2e65d7c217aeb899c11088d648cf89535a89089cdabff9da336ba7e6a739dd5e20a5cf",
+        material
+    );
+    const meshNodes = test.MeshNodes;
+
+    // Create a Scene and insert the MeshNode as the root
+    const scene = new Scene();
+    meshNodes.forEach((node) => scene.addNode(node));
+
+    // Create a Camera to view the scene with
+    const camera = new PerspectiveCamera();
+
+    // Create the renderer
+    const preRender = () => {
+        camera.aspect = gl.canvas.width / gl.canvas.height;
+        meshNodes.forEach((meshNode) => {
+            meshNode.localTransform.rotation = twgl.v3.add(
+                meshNode.localTransform.rotation,
+                [1.0, 1.0, 1.0]
+            );
+        });
+    };
+    const renderer = new Renderer(gl, scene, camera, preRender);
+
+    // Create the game logic handler
+    const game = new Game(map);
+
+    // Create a console log of the FPS
+    let frames = 0;
+    const fpsConsoleLog = setInterval(() => {
+        // eslint-disable-next-line no-console
+        console.log(`FPS: ${frames}`);
+        frames = 0;
+    }, 1000);
+    if (!PRINT_FPS) clearInterval(fpsConsoleLog);
+
+    // Begin rendering
+    let animationRequestId = requestAnimationFrame(function step(time: number) {
+        frames++;
+
+        game.step(time);
+        renderer.render(time);
+
+        animationRequestId = requestAnimationFrame(step);
+    });
```

Comment:
I'd like to get other opinions and research into this decision. My original thought was to allow the Renderer to manage the rendering loop and have the game loop maintain the Scene, changing it as necessary in a game step, which would be reflected in the next frame. 

Our visuals could have animations and would benefit from being in the consistently rendered loop. The game loop is not intended to have a consistent step frequency, as our design is such that game state will only step in response to player input.

However, you could combine these into a single loop and call a function in the Game object that probes for an update, only acting if there is a queued input.

Reply:
That is a good point. Since the game state will only change upon player keyboard/mouse input, I will change the code back to having the requestAnimationFrame in the Renderer and update the game state upon player input instead of checking every frame.

Reply:
With the changes up to 73660e8, the game state only updates when there is a relevant keyboard input (WASD/Arrow Keys). The rendering loop is maintained separate from the game loop.

Reply:
Cool. To be clear, it's definitely possible doing the single loop would be better. We may run into that realization upon implementing further features. This is my first time having done something like this, so I'm open to arguments that provide more guidance. :)